Sources/macro.js

```diff
@@ -705,6 +705,11 @@ export function algo(publicAPI, model, numberOfInputs, numberOfOutputs) {
       if (!model.output[count]) {
         return true;
       }
+
+      if (model.output[count].isDeleted()) {
+        return true;
+      }
+
       const mt = model.output[count].getMTime();
       if (mt < localMTime) {
```

Comment:
@floryst ah, we're comparing `MTime` of `output` and of the actual algo, not of the `input`, so that makes sense, since I assume the modified flag will be changed when the input is changed on the algo. I believe that covers the concern I had.